runtime/vm/createramclass.cpp

```diff
@@ -2877,19 +2956,43 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 			}
 		}
 	}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	if (romFieldCount > DEFAULT_SIZE_OF_CLASS_CACHE) {
+		flattenedClassCache = (J9FlattenedClassCache *) j9mem_allocate_memory(sizeof(J9FlattenedClassCache) * (romFieldCount + 1), J9MEM_CATEGORY_CLASSES);
```

Comment:
Maybe put the alloc/zero size in to a local.

Reply:
Is the size (fieldCount + 1) used other than to be mutitplied? If not, the full value may as well be in the local.

Reply:
I have moved the full value to the local